src/components/checkbox/checkbox.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  Input,
+  Output,
+  Provider,
+  Renderer,
+  ViewEncapsulation,
+  forwardRef
+} from 'angular2/core';
+
+import {
+  NG_VALUE_ACCESSOR,
+  ControlValueAccessor
+} from 'angular2/src/common/forms/directives/control_value_accessor';
+import {CONST_EXPR} from 'angular2/src/facade/lang';
+
+import { OneOf } from '../../core/annotations/one-of';
+
+/**
+ * Monotonically increasing integer used to auto-generate unique ids for checkbox components.
+ */
+let nextId = 0;
+
+/**
+ * Provider Expression that allows md-checkbox to register as a ControlValueAccessor. This allows it
+ * to support [(ngModel)] and ngControl.
+ */
+const MD_CHECKBOX_CONTROL_VALUE_ACCESSOR = CONST_EXPR(new Provider(
+    NG_VALUE_ACCESSOR, {
+      useExisting: forwardRef(() => MdCheckbox),
+      multi: true
+    }));
+
+/**
+ * Represents the different states that require custom transitions between them.
+ */
+enum TransitionCheckState {
+  /** The initial state of the component before any user interaction. */
+  Init,
+  /** The state representing the component when it's becoming checked. */
+  Checked,
+  /** The state representing the component when it's becoming unchecked. */
+  Unchecked,
+  /** The state representing the component when it's becoming indeterminate. */
+  Indeterminate
+}
+
+/**
+ * A material design checkbox component. Supports all of the functionality of an HTML5 checkbox,
+ * and exposes a similar API. An MdCheckbox can be either checked, unchecked, indeterminate, or
+ * disabled. Note that all additional accessibility attributes are taken care of by the component,
+ * so there is no need to provide them yourself. However, if you want to omit a label and still
+ * have the checkbox be accessible, you may supply an [aria-label] input.
+ * See: https://www.google.com/design/spec/components/selection-controls.html#selection-controls-checkbox
+ */
+@Component({
+  selector: 'md-checkbox',
+  templateUrl: './components/checkbox/checkbox.html',
+  styleUrls: ['./components/checkbox/checkbox.css'],
+  host: {
+    'role': 'checkbox',
+    '[id]': 'id',
+    '[class.md-checkbox]': 'true',
+    '[class.md-checkbox-indeterminate]': 'indeterminate',
+    '[class.md-checkbox-checked]': 'checked',
+    '[class.md-checkbox-disabled]': 'disabled',
+    '[class.md-checkbox-align-end]': 'align == "end"',
+    '[tabindex]': 'disabled ? -1 : tabindex',
+    '[attr.aria-label]': 'ariaLabel',
+    '[attr.aria-labelledby]': 'labelId',
+    '[attr.aria-checked]': 'getAriaChecked()',
+    '[attr.aria-disabled]': 'disabled',
+    '(click)': 'onInteractionEvent($event)',
+    '(keyup.space)': 'onInteractionEvent($event)',
+    '(blur)': 'onTouched()'
+  },
+  providers: [MD_CHECKBOX_CONTROL_VALUE_ACCESSOR],
+  encapsulation: ViewEncapsulation.None
```

Comment:
You should be able to add 
```
changeDetection: ChangeDetectionStrategy.OnPush,
```
with everything still working since all of the inputs are primitives